lib/backends/cpu/ops/instance-normalization.ts

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import {InstanceNormalization} from '../../../ops/instance-normalization';
+import {Tensor} from '../../../tensor';
+import {CpuInferenceHandler} from '../inference-handler';
+
+export class CpuInstanceNormalization extends InstanceNormalization {
+  run(inferenceHandler: CpuInferenceHandler, inputs: Tensor[]): Tensor[] {
+    const output = instanceNormalization(inputs[0], inputs[1], inputs[2], this.epsilon);
+    return [output];
+  }
+}
+
+export function instanceNormalization(x: Tensor, scale: Tensor, b: Tensor, epsilon: number) {
+  const inputDimensions = x.dims;
+  const N = inputDimensions[0];
+  const C = inputDimensions[1];
+
+  // calculate channel size (i.e.) data points per channel
+  let channelSize = 1;
+  for (let i = 2; i < inputDimensions.length; i++) {
+    channelSize *= inputDimensions[i];
+  }
+  const output = new Tensor(x.dims, x.type);
+
+  const X = x.floatData;
+  const Y = output.floatData;
+  const scaleData = scale.numberData;
+  const bData = b.numberData;
+
+  let temp: number;
```

Comment:
it is better to define those variables inside the for loop.

- no performance concern like objects (`std::vector<>` for example) in C++
- it's more clear that value of those variables is not carried through iterations.

Reply:
I see what you mean, and I will keep this in mind for future. I vote we keep this as is for now unless you have any concerns.

Reply:
we can keep this as is.